core/src/test/java/org/hibernate/ogm/sessionfactory/SessionFactoryInlineTest.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.sessionfactory;
+
+import static junit.framework.TestCase.fail;
+
+import java.util.function.Consumer;
+
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+
+/**
+ * Helps to test {@link SessionFactory} creation
+ * providing an inline test API
+ *
+ * @author Fabio Massimo Ercoli
+ */
+public class SessionFactoryInlineTest {
+
+	/**
+	 * Expect {@link Exception} during the build of {@link SessionFactory}
+	 *
+	 * @param builder given test configuration
+	 * @param consumer function to apply to the expected exception
+	 */
+	public static void insideCatchException(SessionFactoryConfig builder, Consumer<Exception> consumer) {
+		try ( SessionFactory sessionFactory = builder.getConfiguration().buildSessionFactory( builder.getServiceRegistry() ) ) {
+			fail( "Expected exception at Hibernate factory creation time was not raised!" );
+		}
+		catch (Exception rootException) {
+			consumer.accept( rootException );
+		}
+	}
+
+	/**
+	 * Expect creation of {@link SessionFactory}
+	 *
+	 * @param builder given test configuration
+	 * @param consumer function to apply to the created session factory
+	 */
+	public static void withInSessionFactory(SessionFactoryConfig builder, Consumer<SessionFactory> consumer) {
+		try ( SessionFactory sessionFactory = builder.getConfiguration().buildSessionFactory( builder.getServiceRegistry() ) ) {
+			consumer.accept( sessionFactory );
+		}
+	}
+
+	public static void inTransaction(SessionFactory sessionFactory, Consumer<Session> consumer) {
```

Comment:
Doesn't ORM already have this sort of helper? I recall @DavideD using them.

Reply:
Good question. Hibernate ORM already has, but in its private test module. We already use it a non static within `OgmTestCase` and `OgmJpaTestCase`. In the future we will have all this classes in a dedicated test module, see the issue: https://hibernate.atlassian.net/browse/OGM-1487.